conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanBuilderGenerator.java

```diff
@@ -340,11 +341,10 @@ private CodeBlock typeAwareAssignment(EnrichedField enriched, Type type, boolean
                     spec.name, enriched.fieldName().get() + " cannot be null");
 
             if (isWidenableContainedType(optionalType.getItemType())) {
-                // covariant optionals need to be narrowed to invariant type before assignment
-                Type innerType = optionalType.getItemType();
+                // we need to capture covariant type before assignment to invariant inner variable
                 return CodeBlock.builder()
-                        .addStatement("this.$1N = ($3T<$4T>) $2L",
-                                spec.name, nullCheckedValue, Optional.class, typeMapper.getClassName(innerType).box())
+                        .addStatement("this.$1N = $2L.map($3T.identity())",
```

Comment:
could you also leave a comment here about why we want a `.map(Function.identity())`?

Reply:
Yeah - did you see my update to the comment above (https://github.com/palantir/conjure-java/pull/538/files#diff-3c6ac0d96f3bf3b56cf2702e890c6a96R344) ? Think that captures the reasoning, but happy to move it down to line 346